modules/programs/zed-editor.nix

```diff
@@ -0,0 +1,89 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.zed-editor;
+  jsonFormat = pkgs.formats.json { };
+
+  mergedSettings = cfg.userSettings // {
+    # this part by @cmacrae
+    auto_install_extensions = lib.listToAttrs
+      (map (ext: lib.nameValuePair ext true) cfg.extensions);
+  };
+in {
+  meta.maintainers = [ maintainers.libewa ];
+  options = {
+    programs.zed-editor = {
+      enable = mkEnableOption
+        "Zed, the high performance, multiplayer code editor from the creators of Atom and Tree-sitter";
+      package = mkOption {
+        type = types.package;
+        default = pkgs.zed-editor;
+        defaultText = literalExpression "pkgs.zed-editor";
+        description = ''
+          Another package to install instead of zed
+        '';
+      };
+      userSettings = mkOption {
+        type = jsonFormat.type;
+        default = { };
+        example = literalExpression ''
+          {
+            "features": {
+              "copilot": false
+            },
+            "telemetry": {
+              "metrics": false
+            },
+            "vim_mode": false,
+            "ui_font_size": 16,
+            "buffer_font_size": 16
+          }
+        '';
+        description = ''
+          Configuration written to Zed's {file}`settings.json`.
+        '';
+      };
+      userKeymaps = mkOption {
+        type = jsonFormat.type;
+        default = { };
+        example = literalExpression ''
+          [
+            {
+              "context" = "Workspace";
+              "bindings" = {
+                "ctrl-shift-t" = "workspace::NewTerminal";
+              };
+            };
+          ]
```

Comment:
same as @uncenter 's comment

Reply:
this looks weird because the keys are encased in `"`, but unlike the other example, this is actually valid nixlang (however, this is not how you write it in most cases, so i'm changing that too